sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/UseReturnStatement.vb

```diff
@@ -0,0 +1,23 @@
+﻿﻿Public Class UseReturnStatement
+
+    Public Function FunctionName() As Integer
+    FunctionName = 42 ' Noncompliant {{Use a 'Return' statement; assigning returned values to function names is obsolete.}}
+'       ^^^^^^^^^^^^
+End Function
+
+Public Function ReturnStatement() As Integer
+    Return 42 ' Compliant
+End Function
+
+Public Function ReturnWithVariable() As Integer
+    Dim value As Integer = 42
+    Return value  ' Compliant
+End Function
+
+
+Public Function ReturnWithFunctionName() As Integer
+    ReturnWithFunctionName = 42 ' Noncompliant
+    Return ReturnWithFunctionName  ' Noncompliant
+End Function
+
```

Comment:
Let's add more test cases:
* Compliant recursion
* Noncompliant recursion
* Method reading and writing to a property from another class with the same name as current method name
* Method invoking method from another class with the same name as current method name
* New With property with the same name `Dim Other As New OtherType With {.Something = 42}
* Invoke method with the same name from another class using `With`
```
    Public Function Something(Arg As OtherType) As Integer
        With Arg
            .Something()
        End With
    End Function
```

Reply:
@pavel-mikula-sonarsource  Added those. (And did a rebase).

Reply:
This point is mising:
>Method reading and writing to a property from another class with the same name as current method name